docs/current_docs/features/shell.mdx

```diff
@@ -0,0 +1,94 @@
+---
+slug: /features/shell
+description: "Iterate faster with familiar Bash-like syntax and autocomplete"
+---
+
+# Interactive Shell
+
+Dagger Shell is the fastest way to interact with the Dagger API. Dagger Shell brings the power of pipelines to fully typed API objects while maintaining a familiar Bash-like syntax. With autocomplete for core types, modules, and functions, it makes writing and exploring Dagger pipelines smoother than ever.
+
+Dagger Shell enables you to compose pipelines on the fly, by chaining together core Dagger primitives like `Container` and `Directory` with module-defined types and functions. It can be used inline, with scripts, or interactively. It is secure by design, as all commands execute in containers and you have complete control over what goes in or out from the host system.
+
+Here's an example of Dagger Shell in action:
+
+![Dagger Shell](/img/current_docs/features/shell-curl.gif)
+
+Here is another, more complex example:
+
+![Dagger Shell](/img/current_docs/features/shell-build.gif)
+
+:::note
+Dagger Shell allows you to execute Dagger Functions as native shell commands, using a Bash-compatible syntax. However, instead of calling a subprocess for execution, Dagger Shell directly interprets commands within the Dagger CLI and executes them in containers. This means that all your commands execute as API calls in a secure, sandboxed environment, rather than as processes with direct access to the host system.
+:::
+
+## Modes of execution
+
+Dagger Shell supports multiple ways to run commands:
+
+<Tabs groupId="shell">
+<TabItem value="Inline execution">
+```shell
+dagger -c 'container | from alpine | terminal'
+```
+</TabItem>
+<TabItem value="Standard input">
+```shell title="Type 'dagger' to start Dagger Shell in interactive mode and enter the command below."
+echo 'container | from alpine | terminal' | dagger
+```
+</TabItem>
+<TabItem value="Script">
+```shell
+dagger script.sh
```

Comment:
Is there an example of this script syntax?

Reply:
What do you mean? It's the same as what you pass in `-c` or stdin, just provided as a file.

Reply:
If you want a self-contained example:
```shell
echo 'container | from alpine | with-exec cat /etc/os-release | stdout' > script.sh
dagger script.sh
```